pkg/collector/corechecks/network-devices/cisco-sdwan/report/metrics.go

```diff
@@ -234,6 +234,22 @@ func (ms *SDWanSender) SendDeviceStatusMetrics(deviceStatus map[string]float64)
 	}
 }
 
+// SendHardwareMetrics sends hardware metrics
+func (ms *SDWanSender) SendHardwareMetrics(hardwareEnvironments []client.HardwareEnvironment) {
+	for _, entry := range hardwareEnvironments {
+		devIndex := fmt.Sprintf("%d", entry.HwDevIndex)
+
+		tags := ms.getDeviceTags(entry.VmanageSystemIP)
+		tags = append(tags, "status:"+entry.Status, "class:"+entry.HwClass, "item:"+entry.HwItem, "dev_index:"+devIndex)
+
+		status := 0
```

Comment:
What are the other possible statuses? Is that unbounded?

We usually have four options, one is
`cisco_sdwan.hardware.status.<up|down>`, tagged by the device and with a value of 1 all the time.

One other is similar, but instead of sparse metric, we always emit the metric for every device. If the device is up, we use `1` as a value, and `0` on other status.

One other is
`cisco_sdwan.hardware.status`, tagged by the status and with a value of 1 all the time.

The last is
`cisco_sdwan.hardware.status`, **not tagged** by the status, but with a value of 1 for up and 0 for down. This only works when there is a binary value state

---
Looks like we're doing something in between here with both a status tag, and a value of 1/0. What are the reasons for it?

Reply:
AFAIK the two possible statuses are "OK" and "Down", though we might want to consider the possibility that there's more indeed (here, any other status will be considered as "down")
Is there a specific issue with sending a metric with 0/1 as a value and tagged by `status`? I think it has several advantages but might be missing something
It's easy to filter by the status in widgets, and also easy to set up a monitor using a threshold